piet-gpu/src/lib.rs

```diff
@@ -401,6 +412,19 @@ impl Renderer {
         cmd_buf.end_debug_label();
         cmd_buf.write_timestamp(&query_pool, 1);
         cmd_buf.memory_barrier();
+        cmd_buf.begin_debug_label("Clip bounding box calculation");
+        self.clip_binding
+            .record(cmd_buf, &self.clip_code, self.n_clip as u32);
+        cmd_buf.end_debug_label();
```

Comment:
There is no memory barrier after the refactor. Probably OK, but might want to fix when accounting for the timestamp queries.

Reply:
The lack of memory barrier is deliberate, though maybe could be highlighted in a comment. The convention for the clip dispatch is "barrier in, barrier out." The reason that's significant is that if there are no clips in the scene, there are no dispatches, hence no barriers.

I'll want to redo the timer queries somewhat, as the operations are reordered and there's at least one, perhaps two, additional stages that should get counted. That can happen in this PR or later (I plan to do a bunch of cleanup after the main architectural changes settle).